lib/src/Base/Func/FiniteBasis.cxx

```diff
@@ -43,7 +43,19 @@ FiniteBasis::FiniteBasis(const FunctionCollection & collection)
   : BasisImplementation()
   , collection_(collection)
 {
-  // Nothing to do
+  const UnsignedInteger size = collection.getSize();
+  if (size > 1)
+  {
+    const UnsignedInteger inputDimension = collection[0].getInputDimension();
+    const UnsignedInteger outputDimension = collection[0].getOutputDimension();
+    for (UnsignedInteger i = 1; i < size; ++ i)
+    {
+      if (collection[i].getInputDimension() != inputDimension)
+        throw InvalidArgumentException(HERE) << "Basis elements cannot have different input dimensions";
+      if (collection[i].getOutputDimension() != outputDimension)
+        throw InvalidArgumentException(HERE) << "Basis elements cannot have different output dimensions";
+    }
+  }
```

Comment:
Good
Maybe we should have specific accessors ?
The `Basis::getDimension` is confusing

Reply:
yes, lets add getInputDimension and getOutputDimension and deprecate getDimension